iter.go

```diff
@@ -87,6 +87,16 @@ func (i *Iterator) SeekLowerBound(key []byte) {
 		i.stack = append(i.stack, edges{edge{node: n}})
 	}
 
+	findMin := func(n *Node) {
+		n = i.recurseMin(n)
+		if n != nil {
+			found(n)
+			return
+		}
+		i.node = nil
```

Comment:
Would it be worth adding a comment to the top of the function explaining how `i.node` and `i.stack` should be affected by the different outcomes that may occur?  e.g.:stack will be traversed by Next.  Since stack is non-nil, node will not be used by Next, only by future Seek calls.  We set node = nil when we want to indicate that that there is no lower bound.

I'm actually not clear why we set node=nil - isn't leaving stack empty enough?  It feels like an extra side effect that's not very obvious to the user.  Is the idea that someone would do a Seek, consume the results using Next, then do another Seek from that point using the same iterator?

Maybe we should provide an accessor in Iterator to get the current Node setting, so they can see this outcome more directly?

Reply:
We should also consider validating the Node update side-effect in tests.